contracts/contracts/vault/VaultAdmin.sol

```diff
@@ -0,0 +1,323 @@
+     *                 For ex. 100%=1e18, 30%=3e17.
+     */
+    function setStrategyWeights(
+        address[] calldata _strategyAddresses,
+        uint256[] calldata _weights
+    ) external onlyGovernor {
+        require(
+            _strategyAddresses.length == _weights.length,
+            "Parameter length mismatch"
+        );
+
+        for (uint256 i = 0; i < _strategyAddresses.length; i++) {
+            strategies[_strategyAddresses[i]].targetWeight = _weights[i];
+        }
+
+        emit StrategyWeightsUpdated(_strategyAddresses, _weights);
+    }
+
+    /***************************************
+                    Pause
+    ****************************************/
+
+    /**
+     * @dev Set the deposit paused flag to true to prevent rebasing.
+     */
+    function pauseRebase() external onlyGovernor {
+        rebasePaused = true;
+    }
+
+    /**
+     * @dev Set the deposit paused flag to true to allow rebasing.
+     */
+    function unpauseRebase() external onlyGovernor {
+        rebasePaused = false;
+    }
+
+    /**
+     * @dev Set the deposit paused flag to true to prevent deposits.
+     */
+    function pauseDeposits() external onlyGovernor {
+        depositPaused = true;
+
+        emit DepositsPaused();
+    }
+
+    /**
+     * @dev Set the deposit paused flag to false to enable deposits.
+     */
+    function unpauseDeposits() external onlyGovernor {
+        depositPaused = false;
+
+        emit DepositsUnpaused();
+    }
+
+    /***************************************
+                    Rewards
+    ****************************************/
+
+    /**
+     * @dev Transfer token to governor. Intended for recovering tokens stuck in
+     *      contract, i.e. mistaken sends.
+     * @param _asset Address for the asset
+     * @param _amount Amount of the asset to transfer
+     */
+    function transferToken(address _asset, uint256 _amount)
+        external
+        onlyGovernor
+    {
+        IERC20(_asset).transfer(governor(), _amount);
+    }
+
+    /**
+     * @dev Collect reward tokens from all strategies and swap for supported
+     *      stablecoin via Uniswap
+     */
+    function harvest() external onlyGovernor {
+        for (uint256 i = 0; i < allStrategies.length; i++) {
+            _harvest(allStrategies[i]);
+        }
+    }
+
+    /**
+     * @dev Collect reward tokens for a specific strategy and swap for supported
+     *      stablecoin via Uniswap
+     * @param _strategyAddr Address of the strategy to collect rewards from
+     */
+    function harvest(address _strategyAddr) external onlyGovernor {
+        _harvest(_strategyAddr);
+    }
+
+    /**
+     * @dev Collect reward tokens from a single strategy and swap them for a
+     *      supported stablecoin via Uniswap
+     * @param _strategyAddr Address of the strategy to collect rewards from
+     */
+    function _harvest(address _strategyAddr) internal {
```

Comment:
Should this method emit an event for each harvested token including the token address and amount. Though I assume Uniswap emits events as well ? in that case this could be redundant

Reply:
The strategy emits an event like that, so that is probably enough to cover us.